projects/core/src/lib/users/user.service.ts

```diff
@@ -49,19 +49,24 @@ export class UserService {
 
             if (status.available && !this.sessionInitialized) {
                 this.sessionInitialized = true;
-                // restore old session if possible
-                this.sessionFromBrowserOrCreateGuest().subscribe({
-                    next: (session) => {
-                        this.session$.next(session);
-                    },
-                    error: (error) => {
-                        // only show error if we did not expect it
-                        if (error.error.error !== 'AnonymousAccessDisabled') {
-                            this.notificationService.error(error.error.message);
-                        }
-                        this.session$.next(undefined);
-                    },
-                });
+
+                const oidcLogin = this.checkOidcParametersAndLogin();
```

Comment:
Don't you need to await something?

Reply:
no. This methods checks the URL parameters which do not have to be awaited. in case they are set, the oids login happens asynchronously and false is returned so the regular login is skipped. If the parameters are not present, the login works like before.

Reply:
refactor like discussed in chat